src/halmos/sevm.py

```diff
@@ -585,6 +585,53 @@ def __next__(self) -> Instruction:
         return insn
 
 
+class Path:
```

Comment:
nit: would you say that an instance of this class does represent a single path through the program? My instinct says no, because it makes it a little weird to initialize an SEVM with a Path object (since I don't really have a path before we get started). 

Semantically, would something like `PathTracker` or something similar be more accurate?

Reply:
fair point!  but in my mental model, it's still a path.  i added explanation in the comment for now.  we can discuss more to improve naming.